solver/src/main/java/org/chocosolver/solver/constraints/set/PropUnionVar.java

```diff
@@ -70,6 +68,20 @@ public PropUnionVar(SetVar union, int vOffset, SetVar indices, int iOffset, SetV
     public void propagate(int evtmask) throws ContradictionException {
         SetVar union = vars[k];
         SetVar indices = vars[k + 1];
+        if (indices.getUB().size() > 0) {
+            int min = indices.getUB().min();
+            while (min - iOffset < 0 && indices.getUB().size() > 0) {
+                indices.remove(min, this);
+                min = indices.getUB().min();
+            }
+        }
+        if (indices.getUB().size() > 0) {
+            int max = indices.getUB().max();
+            while (max - iOffset > k && indices.getUB().size() > 0) {
```

Comment:
>= k ?

Reply:
`k` is equal to sets.length I'll change that

Reply:
I won't change, this is a class variable